serviceregistration/consul/consul_service_registration.go

```diff
@@ -178,47 +213,33 @@ func NewServiceRegistration(conf map[string]string, logger log.Logger, state sr.
 	}
 
 	if consulConf.Scheme == "https" {
+
```

Comment:
I think this operation is quick enough to do no matter what.  Then if we like the otel approach you just call diagnose.Error if it fails.

Reply:
It's definitely quick enough, but if the user doesn't specify tls wouldn't they be surprised by the diagnose TLS error?